ql/src/java/org/apache/hadoop/hive/ql/DriverFactory.java

```diff
@@ -49,20 +47,29 @@ public static IDriver newDriver(QueryState queryState, QueryInfo queryInfo) {
     }
 
     String strategies = queryState.getConf().getVar(ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES);
-    strategies = Strings.nullToEmpty(strategies).trim().toLowerCase();
     List<IReExecutionPlugin> plugins = new ArrayList<>();
     for (String string : strategies.split(",")) {
       if (string.trim().isEmpty()) {
         continue;
       }
-      plugins.add(buildReExecPlugin(string));
+
+      IReExecutionPlugin plugin = buildReExecPlugin(string);
+      if (plugin != null) {
+        plugins.add(buildReExecPlugin(string));
+      } else {
+        plugins.add(buildCustomReExecPlugin(string));
+      }
     }
 
     return new ReExecDriver(queryState, queryInfo, plugins);
   }
 
   private static IReExecutionPlugin buildReExecPlugin(String name) throws RuntimeException {
     Class<? extends IReExecutionPlugin> pluginType = ReExecutionStrategyType.getPluginClassByName(name);
+    if (pluginType == null) {
```

Comment:
why not catch  exception and do a fallback here

````
    Class<?> pluginType;
    try {
      pluginType = ReExecutionStrategyType.getPluginClassByName(name);
    } catch (IllegalArgumentException e) {
      pluginType = Class.forName(name);
      if (!pluginType.isAssignableFrom(IReExecutionPlugin.class)) {
        throw new RuntimeException("Not re-execution plugin: " + name);
      }
    }
    try {
      return (IReExecutionPlugin) pluginType.newInstance();
    } catch (InstantiationException | IllegalAccessException e) {
      throw new RuntimeException(
          "Unknown re-execution plugin: " + name + " (" + ConfVars.HIVE_QUERY_REEXECUTION_STRATEGIES.varname + ")");
    }
````

Reply:
OK, move the process.